### PR TITLE
Trim AMYDeleteView of any redirection support

### LINKED ISSUE #1097: Incorrect redirect on deleting an award from person detail page
On the `person_detail` page, deleting an award from the _Awards_ tab takes you to the `badge_detail` page.

Should it be redirected back to the `person_detail` page?

### LINKED ISSUE #1101: AMYDeleteView and RedirectSupportMixin achieve different redirection
The `next` param to the `AMYDeleteView` is a POST param while to any view subclassing the `RedirectSupportMixin` mixin, the `next` param is a GET param. The two achieve the same thing, but differently.

I propose that we strip the `AMYDeleteView` of any redirect support and use `RedirectSupportMixin` along with `AMYDeleteView`.

### PR DESCRIPTION
**Pull request by narayanaditya95**
Fixes #1097
Fixes #1101

### CODE REVIEW

File: workshops/views.py [hunk 1]
```diff
@@ -1587,12 +1587,16 @@ class TaskDelete(OnlyForAdminsMixin, PermissionRequiredMixin,
...
     model = Award
     permission_required = 'workshops.delete_award'
 
     def get_success_url(self):
-        return reverse('person_edit', args=[self.get_object().person.pk]) + '#awards'
+        next_url = self.request.GET.get('next', None)
+        if next_url:
+            return next_url
+        else:
+            return reverse('badge_details', args=[self.get_object().badge.name])
```
Comment (narayanaditya95) [inline workshops/views.py]: I had to replicate the entire functionality of `RedirectSupportMixin`. Is there a simpler way out?
@pbanaszkiewicz @chrismedrela

Comment (pbanaszkiewicz) [inline workshops/views.py]:
```python
class MockAwardDelete(OnlyForAdminsMixin, PermissionRequiredMixin,
                      AMYDeleteView):
    model = Award
    permission_required = 'workshops.delete_award'

    def get_success_url(self):
        return reverse('badge_details', args=[self.get_object().badge.name])


class AwardDelete(RedirectSupportMixin, MockAwardDelete):
    pass
```

This changes the method resolution order, but I think it should work in our case.

Comment (narayanaditya95) [inline workshops/views.py]: @pbanaszkiewicz That worked :)

**Comment by narayanaditya95** [comment]
@pbanaszkiewicz Ready to be merged.

File: workshops/views.py [hunk 2]
```diff
@@ -1586,13 +1586,17 @@ class TaskDelete(OnlyForAdminsMixin, PermissionRequiredMixin,
...
     model = Award
     permission_required = 'workshops.delete_award'
 
     def get_success_url(self):
-        return reverse('person_edit', args=[self.get_object().person.pk]) + '#awards'
+        return reverse('badge_details', args=[self.get_object().badge.name])
+
+
+class AwardDelete(RedirectSupportMixin, MockAwardDelete):
+    pass
```
Comment (pbanaszkiewicz) [inline workshops/views.py]: I'd like a comment here what this does and why :-)

Comment (narayanaditya95) [inline workshops/views.py]: Done

**Comment by pbanaszkiewicz** [comment]
Thanks @narayanaditya95 !

**Pull request merged**